### ISSUE TITLE
[Enhancement] Improve performance, when huge block of code is edited

### ISSUE THREAD

**Issue by subhamayd2** [BertrandBev/code_field#10]
When I am editing a huge block of code, the editor feels a bit sluggish.
This is probably happening because on each change, the whole block of code is being parsed for highlighting.
Is there any way that can improve the performance? I was thinking, if the parsing can be debounced somehow, that might improve the performance.

**Comment by BertrandBev** [comment]
Yep, we could also perform local highlighting, since most of the time local changes don't affect the rest of the document!

**Comment by subhamayd2** [comment]
Agreed. Something like, only parsing the line that is currently being edited.

**Comment by alberteije** [comment]
Any improvement on this? Looks like we have same situation until now.
